src/CoreBundle/private/library/classes/TCMSResourceCollection.class.php

```diff
@@ -69,98 +81,161 @@ public function IsEnabled()
-                $sReworkContent = preg_replace_callback($matchString, array($this, 'CollectExternalResourcesJSCallback'), $sReworkContent);
-                $aJS = $this->StaticContentCollector('js');
-                $minify = ServiceLocator::getParameter('chameleon_system_core.resources.enable_external_resource_collection_minify');
-                if ($minify) {
-                    $sMinifyStatus = 'true';
-                } else {
-                    $sMinifyStatus = 'false';
-                }
-                $sFileJSMD5 = $filesPrefix.'.js.'.md5(implode(';', $aJS).$sMinifyStatus).'.js';
-                if (false === $this->CreateJSResourceCollectionFile($sFileJSMD5, $aJS)) {
-                    $sFileJSMD5 = ''; // reset file because we don`t have includes
-                }
+        $hasCollectedCss = \count($aCSS) > 0 && \count($aGlobalCSS);
+        $hasCollectedJs = \count($aJS) > 0 && \count($aJSGlobal);
+        if (false === $hasCollectedCss && false === $hasCollectedJs) {
+            return $sPageContent;
+        }
 
-                $matchString = "/<script([^>]+?)src=[\"]([^'\"]+?).js([\?][^[:space:]]+)?[\"]([^>]*?)><\/script>\\s*(<!--.*?#GLOBALRESOURCECOLLECTION#.?-->)/i";
-                $sReworkContent = preg_replace_callback($matchString, array($this, 'CollectExternalResourcesJSCallback'), $sReworkContent);
-                $aJSGlobal = $this->StaticContentCollector('jsglobal');
-                $sFileJSGlobalMD5 = $filesPrefix.'.global_js.'.md5(implode(';', $aJSGlobal).$sMinifyStatus).'.js';
-                if (false === $this->CreateJSResourceCollectionFile($sFileJSGlobalMD5, $aJSGlobal)) {
-                    $sFileJSGlobalMD5 = ''; // reset global file because we don`t have global includes
-                }
+        $sFileCSSMD5 = $filesPrefix.'.css.'.md5(implode(';', $aCSS)).'.css';
+        $sFileJSMD5 = $filesPrefix.'.js.'.md5(implode(';', $aJS).$sMinifyStatus).'.js';
+        $sFileCSSGlobalMD5 = $filesPrefix.'.global_css.'.md5(implode(';', $aGlobalCSS)).'.css';
+        $sFileJSGlobalMD5 = $filesPrefix.'.global_js.'.md5(implode(';', $aJSGlobal).$sMinifyStatus).'.js';
+        $resourceFilesExist = \file_exists($sFileCSSMD5);
+        $resourceFilesExist = $resourceFilesExist && \file_exists($sFileJSMD5);
+        $resourceFilesExist = $resourceFilesExist && \file_exists($sFileCSSGlobalMD5);
+        $resourceFilesExist = $resourceFilesExist && \file_exists($sFileJSGlobalMD5);
 
-                if (2 == count($aPageParts)) {
-                    $sPageContent = $sReworkContent.'</head>'.$aPageParts[1];
-                } else {
-                    $sPageContent = $sReworkContent;
-                }
-                $sPageContent = $this->AddResourceCollectionToHead($sPageContent, $sFileCSSMD5, $sFileJSMD5, $sFileCSSGlobalMD5, $sFileJSGlobalMD5);
+        if (false === $resourceFilesExist) {
```

Comment:
It needs to check the lock here before setting it. (?)

Reply:
not sure what you mean. We check if all files exist - if some don't, then files will be written and we need a lock. If all files exists, no files will be written so we need no lock

Reply:
lock is now checked as part of the lock acquire method

Reply:
The three methods below would also make this here more clear:

if "is running" return early.

(otherwise) start

Reply:
I don't understand what you mean

Reply:
ok - I understand now... The logic hier is: attempt to acquire lock - if that fails, do not run the resource collection. 

Acquiring the lock will fail if someone already has the lock - but also when we fail to write the lock file. So if we moved to the three methods you suggested, then the logic here would become even more complex (because we would still need to check if we succeeded to acquire the lock).